libs/FirmwareKit/tests/FirmwareKit_test.cpp

```diff
@@ -73,23 +52,60 @@ TEST_F(FirmwareKitTest, getCurrentVersion)
 
 TEST_F(FirmwareKitTest, loadUpdate)
 {
+	auto version_to_load = Version {1, 0, 0};
 	{
 		InSequence seq;
 
 		EXPECT_CALL(mock_flash, erase).Times(1);
-		EXPECT_CALL(mock_flash, write(_, compareArray(bin_update_content, _), std::size(bin_update_content))).Times(1);
+		EXPECT_CALL(mock_flash, write).Times(AtLeast(1));
 	}
 
-	auto did_load_firmware = firmwarekit.loadUpdate(update_version);
+	auto did_load_firmware = firmwarekit.loadUpdate(version_to_load);
 
 	ASSERT_TRUE(did_load_firmware);
 }
 
 TEST_F(FirmwareKitTest, loadUpdateFileNotFound)
 {
-	std::filesystem::remove(bin_update_path.c_str());
+	auto unexisting_version = Version {0, 0, 0};
 
-	auto did_load_firmware = firmwarekit.loadUpdate(update_version);
+	auto did_load_firmware = firmwarekit.loadUpdate(unexisting_version);
 
 	ASSERT_FALSE(did_load_firmware);
 }
+
+TEST_F(FirmwareKitTest, isVersionAvailable)
+{
+	auto existing_version	  = Version {1, 0, 0};
+	auto is_version_available = firmwarekit.isVersionAvailable(existing_version);
+
+	EXPECT_TRUE(is_version_available);
+}
+
+TEST_F(FirmwareKitTest, isVersionAvailableFileNotFound)
+{
+	auto unexisting_version = Version {0, 0, 0};
+
+	auto is_version_available = firmwarekit.isVersionAvailable(unexisting_version);
+
+	EXPECT_FALSE(is_version_available);
+}
+
+TEST_F(FirmwareKitTest, isVersionAvailableFileTooSmall)
+{
+	auto dummy_version				   = Version {99, 99, 9999};
+	std::string bin_dummy_version_path = "fs/usr/os/LekaOS-99.99.9999.bin";
```

Comment:
the test is great, but I'd prefer we don't use `fs/` for creating files.

if for some reason the test crashes before removing at the end, you'll end up with new files in `fs/` and there is a risk someone might commit those by accident.

It would be better to use the `tmp/` directory for this case, even though it means instantiating a new firmwarekit object